src/ui/components/Difference/index.jsx

```diff
@@ -2,40 +2,130 @@ import React, { useState, useEffect } from "react";
 import styled from "styled-components";
 import { nanoid } from "nanoid";
 
+import { useNavigate } from "react-router-dom";
 import Description from "../shared/Description";
 import Button from "../shared/Button";
-import Modal from "../shared/Modal";
+import ToastPopup from "../shared/Toast";
 import Loading from "../shared/Loading";
+import Modal from "../shared/Modal";
 import Popup from "../shared/Popup";
 
+import usePageListStore from "../../../store/projectPage";
+import useProjectStore from "../../../store/project";
+
+import isOwnProperty from "../../../utils/isOwnProperty";
+import postMessage from "../../../utils/postMessage";
+import getDiffingResultQuery from "../../../services/getDiffingResultQuery";
 import processDifferences from "../../../utils/processDifferences";
 
 function Difference() {
-  const [isLoading, setIsLoading] = useState(false);
+  const [toast, setToast] = useState({});
+  const [pageId, setPageId] = useState("");
   const [isOpenedPopup, setIsOpenedPopup] = useState(false);
+  const [projectStatus, setProjectStatus] = useState({});
+  const { allPageIds } = usePageListStore();
+  const { project } = useProjectStore();
+
+  const { data: diffingResult, isLoading } = getDiffingResultQuery(
+    projectStatus.projectKey,
+    project.beforeVersionId,
+    project.afterVersionId,
+    pageId,
+    projectStatus.accessToken,
+  );
+  const navigate = useNavigate();
   const [displayText, setDisplayText] = useState({
     titleOfChanges: null,
     detailOfChanges: ["변경사항을 선택해주세요."],
     className: "default",
   });
```

Comment:
- `P2`
```suggestion
  const navigate = useNavigate();
  
  const [toast, setToast] = useState({});
  const [pageId, setPageId] = useState("");
  const [isOpenedPopup, setIsOpenedPopup] = useState(false);
  const [projectStatus, setProjectStatus] = useState({});
  const [displayText, setDisplayText] = useState({
    titleOfChanges: null,
    detailOfChanges: ["변경사항을 선택해주세요."],
    className: "default",
  });

  const { allPageIds } = usePageListStore();
  const { project } = useProjectStore();

  const { data: diffingResult, isLoading } = getDiffingResultQuery(
    projectStatus.projectKey,
    project.beforeVersionId,
    project.afterVersionId,
    pageId,
    projectStatus.accessToken,
  );
```

관련된 로직끼리 묶고, 개행을 나누면 더 보기 좋을 것 같습니다!